lib/socket.c

```diff
@@ -322,6 +324,16 @@ rpc_write_to_socket(struct rpc_context *rpc)
                 char *last_buf = NULL;
                 ssize_t count;
 
+                assert(pdu->out.niov <= pdu->out.iov_capacity);
+                assert(pdu->out.iov_capacity <= RPC_MAX_VECTORS);
+
+                if (pdu->out.niov > iovcnt && iovcnt != RPC_MAX_VECTORS) {
+                        assert(iov == fast_iov);
+                        iov = (struct iovec *) calloc(RPC_MAX_VECTORS,
+                                                      sizeof(struct iovec));
```

Comment:
Check for NULL.  Embedded platforms generally have limited facilities to "just restart the app if malloc failed and it crashed".